web/openapi/openapi.yaml

```diff
@@ -50,6 +50,20 @@ components:
           $ref: "#/components/schemas/PipelineAutostart"
         autodelete:
           $ref: "#/components/schemas/PipelineAutodelete"
+        max_retries:
+          type: integer
+          description: If `autostart.failed` is `true`, the maximum amount of retries
+            for a failed pipeline.
+          default: 10
+          example: 15
+        retry_delay_ns:
```

Comment:
Can we make this a duration? (Nanoseconds are not a user-facing unit in config files.)

Reply:
Yes, we can make this a duration when creating/updating so that the user specifies a duration string, like with the TTL - should we return the duration in nanoseconds when retrieving the pipeline information (just like with the TTL)?

Reply:
> Yes, we can make this a duration

Great.



> should we return the duration in nanoseconds when retrieving the pipeline information (just like with the TTL)

Consistency trumps, so "yes" here as well. There should be only one way to retrieve a duration.

Reply:
I applied the change.

Reply:
Not seeing it yet. Did I miss something?